FlowCryptCommon/Extensions/StringExtensions.swift

```diff
@@ -61,6 +65,36 @@ public extension NSAttributedString {
 
 // MARK: Email parsing
 public extension String {
+    var isValidEmail: Bool {
+        let emailFormat = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+        let emailPredicate = NSPredicate(format: "SELF MATCHES %@", emailFormat)
+        return emailPredicate.evaluate(with: self)
+    }
+    
+    var userAndRecipientDomain: (user: String, domain: String)? {
+        let parts = self.split(separator: "@")
+        if parts.count != 2 {
+            return nil
+        }
+        return (String(parts[0]), String(parts[1]))
+    }
+    
+    var userEmail: String? {
+        let parts = self.split(separator: "@")
+        if parts.count != 2 {
+            return nil
+        }
+        return String(parts[0])
+    }
+    
+    var recipientDomain: String? {
+        let parts = self.split(separator: "@")
+        if parts.count != 2 {
+            return nil
+        }
+        return String(parts[1])
+    }
+    
```

Comment:
`userAndRecipientDomain`, `userEmail` and `recipientDomain` properties were replaced by the single `emailParts` property, so they can be deleted